libr/main/rax2.c

```diff
@@ -55,7 +55,10 @@ static int format_output(RNum *num, char mode, const char *s, int force_mode, ut
 	}
 	if (flags & 2) {
 		ut64 n2 = n;
-		r_mem_swapendian ((ut8 *) &n, (ut8 *) &n2, (n >> 32)? 8: 4);
+		r_mem_swapendian ((ut8 *) &n, (ut8 *) &n2, 8);
+		if (!(int) n) {
```

Comment:
Why the cast

Reply:
If n was 32 bit, after swapping endian its lower 32 bits are zero.